File: nodejs/src/entity/ResearchActivity.ts
import { EntitySchema } from 'typeorm';

export interface ResearchActivity {
    time: Date;
    researcher: string;
    paper: string;
    topic: string;
    conference: string;
    organization: string;
    citations?: number;
}

export const ResearchActivitySchema = new EntitySchema<ResearchActivity>({
    name: "ResearchActivity",
    tableName: "research_activity",
    columns: {
        time: {
            type: Date,
            primary: true,
        },
        researcher: {
            type: String,
            length: 255,
        },
        paper: {
            type: String,
            length: 255,
        },
        topic: {
            type: String,
            length: 255,
        },
        conference: {
            type: String,
            length: 255,
        },
        organization: {
            type: String,
            length: 255,
        },
        citations: {
            type: Number,
            nullable: true,
        },
    },
});

File: golang/models/research_activity.go
package models

import (
	"time"
)

type ResearchActivity struct {
	//Id           uint      `gorm:"primaryKey" json:"id"`
	Time         time.Time `json:"time"`
	Researcher   string    `json:"researcher"`
	Paper        string    `json:"paper"`
	Topic        string    `json:"topic"`
	Conference   string    `json:"conference"`
	Organization string    `json:"organization"`
	Citations    int       `json:"citations"`
}

func (ResearchActivity) TableName() string {
	return "research_activity" // or whatever your exact table name is
}
